### PR TITLE
Inode operations

### PR DESCRIPTION
**Pull request by anton-bondarev**
Add struct inode_operations for oldfs and switch devfs on it

### CODE REVIEW

File: src/compat/posix/fs/dirent/dirent_oldfs.c [hunk 1]
```diff
@@ -61,39 +62,67 @@ DIR *opendir(const char *path) {
...
+			node = inode_new(NULL);
+			if (NULL == node) {
+				SET_ERRNO(ENOMEM);
+				return NULL;
+			}
+			res = dir->path.node->i_ops->iterate(node, dir->path.node, &dir->dir_contex);
+			if (res == -1){
+				node_free(node);
+				return NULL;
+			}
```
Comment (alexkalmuk) [inline src/compat/posix/fs/dirent/dirent_oldfs.c]: Should'nt we also do SET_ERRNO here like above?

File: src/fs/driver/devfs/devfs_oldfs.c [hunk 2]
```diff
@@ -120,12 +123,114 @@ int devfs_add_char(struct dev_module *cdev) {
...
+	/* Dynamically allocated devices */
+	for (; i < (MAX_CDEV_QUANTITY + MAX_BDEV_QUANTITY); i++) {
+		if (cdevtab[i - MAX_BDEV_QUANTITY]) {
+			ctx->fs_ctx = (void *) ((intptr_t) i + 1);
+			devfs_fill_inode(next, cdevtab[i - MAX_BDEV_QUANTITY], S_IFCHR | S_IRALL | S_IWALL);
+			strncpy(next->name, cdevtab[i - MAX_BDEV_QUANTITY]->name, sizeof(next->name) - 1);
+			next->name[sizeof(next->name) - 1] = '\0';
+			return 0;
+		}
+	}
```
Comment (alexkalmuk) [inline src/fs/driver/devfs/devfs_oldfs.c]: May be use `for (i = 0; i < MAX_CDEV_QUANTITY; i++) {` to get rid of `cdevtab[i - MAX_BDEV_QUANTITY]` and just use `cdevtab[i]`?

File: src/fs/driver/devfs/devfs_oldfs.c [hunk 3]
```diff
@@ -120,12 +123,114 @@ int devfs_add_char(struct dev_module *cdev) {
...
+	//struct dev_module *dev_module;
+	struct dev_module **cdevtab = get_cdev_tab();
+
+	i = ((intptr_t) ctx->fs_ctx);
+
+	for (; i < MAX_BDEV_QUANTITY; i++) {
+		if (bdevtab[i]) {
+			ctx->fs_ctx = (void *)(intptr_t)i + 1;
+			devfs_fill_inode(next, bdevtab[i]->dev_module, S_IFBLK | S_IRALL | S_IWALL);
+			//next->length = bdevtab[i]->size;
```
Comment (alexkalmuk) [inline src/fs/driver/devfs/devfs_oldfs.c]: It's better to remove or provide some explanation.

File: src/fs/driver/devfs/devfs_oldfs.c (hunk 2 above)
Comment (alexkalmuk) [reply src/fs/driver/devfs/devfs_oldfs.c]: Ah, now I see we have common counter for both bdev and cdev.

File: src/compat/posix/fs/dirent/dirent_oldfs.c [hunk 4]
```diff
@@ -61,39 +62,67 @@ DIR *opendir(const char *path) {
 
 	d->path = node_path;
 	d->current.d_ino = 0;
+	d->dir_contex.fs_ctx = 0;
```
Comment (alexkalmuk) [inline src/compat/posix/fs/dirent/dirent_oldfs.c]: dir_context?

File: src/fs/driver/devfs/devfs_oldfs.c [hunk 5]
```diff
@@ -120,12 +123,114 @@ int devfs_add_char(struct dev_module *cdev) {
...
+
+/**
+ * @brief Find file in directory
+ *
+ * @param name Name of file
+ * @param dir  Pointer to directory
+ *
+ * @return Pointer to inode structure or NULL if failed
+ */
+static struct inode *devfs_lookup(char const *name, struct dentry const *dir) {
```
Comment (alexkalmuk) [inline src/fs/driver/devfs/devfs_oldfs.c]: Hm, so iterate works for voth block devs and char devs, while lookup only for char devices?

File: src/fs/driver/devfs/devfs_oldfs.c (hunk 3 above)
Comment (anton-bondarev) [reply src/fs/driver/devfs/devfs_oldfs.c]: done

File: src/compat/posix/fs/dirent/dirent_oldfs.c (hunk 4 above)
Comment (anton-bondarev) [reply src/compat/posix/fs/dirent/dirent_oldfs.c]: thanks, done

File: src/fs/driver/devfs/devfs_oldfs.c (hunk 5 above)
Comment (anton-bondarev) [reply src/fs/driver/devfs/devfs_oldfs.c]: done

File: src/compat/posix/fs/dirent/dirent_oldfs.c (hunk 1 above)
Comment (anton-bondarev) [reply src/compat/posix/fs/dirent/dirent_oldfs.c]: it's not a standard, and I don't know what code must be set, I return NULL, and it shows that function failed

**Pull request merged**